doubly_linked_list/doubly_linked_list.py

```diff
@@ -40,10 +40,23 @@ def __init__(self, node=None):
     self.tail = node
 
   def add_to_head(self, value):
-    pass
-
+      node_to_add = ListNode(value)
+      if self.head is None:
+          self.head = node_to_add
+      else:
+          self.head.insert_before(value)
+          self.head = node_to_add
+          
   def remove_from_head(self):
-    pass
+      if self.head is None:
+          return None
+      elif self.head is not None:
+          head_to_delete = self.head
+          if self.head.next is not None:
+              self.head.delete()
+              self.head = self.head.next
+              return head_to_delete.value
+
 
   def add_to_tail(self, value):
     pass
```

Comment:
It looks like you need to work on add_to_tail, remove_from_tail, move_to_front, move_to_end, and delete to get these tests passing.